File: src/doerly-fe/src/app/modules/order/models/requests/create-order-request.ts
import { EPaymentKind } from "../../domain/enums/payment-kind";

export interface CreateOrderRequest {
    categoryId: number;
    name: string;
    description: string;
    price: number;
    isPriceNegotiable: boolean;
    paymentKind: EPaymentKind;
    dueDate: Date;
}

File: src/main-service/Doerly.Module.Order/Doerly.Module.Order.Contracts/Dtos/Requests/Order/CreateOrderRequest.cs
using Doerly.Domain;
using Doerly.Module.Order.Enums;
using System.ComponentModel.DataAnnotations;

namespace Doerly.Module.Order.Contracts.Dtos;

public class CreateOrderRequest
{
    [Required]
    public int CategoryId { get; set; }

    [Required]
    public string Name { get; set; }

    [Required]
    public string Description { get; set; }

    [Required]
    public decimal Price { get; set; }

    [Required]
    public EPaymentKind PaymentKind { get; set; }

    [Required]
    public DateTime DueDate { get; set; }
}
